llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -1213,13 +1218,13 @@ void RISCVInsertVSETVLI::transferBefore(VSETVLIInfo &Info,
 void RISCVInsertVSETVLI::transferAfter(VSETVLIInfo &Info,
                                        const MachineInstr &MI) const {
   if (isVectorConfigInstr(MI)) {
-    Info = getInfoForVSETVLI(MI);
+    Info = getInfoForVSETVLI(MI, *MRI);
     return;
   }
 
   if (RISCV::isFaultFirstLoad(MI)) {
     // Update AVL to vl-output of the fault first load.
-    Info.setAVLReg(MI.getOperand(1).getReg());
+    Info.setAVLDefMI(MRI->getVRegDef(MI.getOperand(1).getReg()));
```

Comment:
Just making a note, I don't think we have any pseudos that have more than one scalar def, but if we did then we would also need to track the operand number.